stock_quant_partner_info/models/stock_quant.py

```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+# © 2016 Oihane Crucelaegui - AvanzOSC
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from openerp import api, fields, models
+
+
+class StockQuant(models.Model):
+    _inherit = 'stock.quant'
+
+    dest_partner_id = fields.Many2one(
+        comodel_name='res.partner', compute='_compute_partner_id',
+        string='Partner', store=True)
+
+    @api.depends('location_id', 'history_ids')
+    def _compute_partner_id(self):
+        for quant in self.filtered(
+                lambda q: q.location_id.usage in ['customer', 'supplier']):
+            move = quant.history_ids.filtered(
```

Comment:
Don't filter again the history_ids by location. You already have it on `[:-1]`, as you are checking previously the location_id usage.